### DISCUSSION TITLE
Increasing the Size of A guestVM

### DISCUSSION THREAD

**Discussion by chiibuzor** [firecracker-microvm/firecracker#3110] [Q&A]

**Comment by dianpopa** [comment]
Hi!

Can you give us more details? What resource of your guestVM would you need to adjust?

**Comment by chiibuzor** [comment]
Hi,

I would like to increase the size of the root partition, if possible mount
another drive with a desired size.

Cheers. Regards

**Comment by dianpopa** [comment]
Yes, you can adjust the size of a block device after boot.

You simply modify the size of the backing file on the host and then you send the patch command for the associated block device.
You can find more details here: https://github.com/firecracker-microvm/firecracker/blob/a367796e66eeac42d9ce1294c0fbbca6191e9cf3/docs/api_requests/patch-block.md.

**Comment by chiibuzor** [comment]
Thanks, I'll check it out.

**Comment by WeeJeWel** [comment]
Is there also something documented, so that the (.ext4) file on the host only takes up what's actually inside, but can grow up to a certain limit? Similar to how many popular VM solutions do it (e.g. VMWare, Parallels).

**Comment by ShadowCurse** [comment]
Hi @WeeJeWel, Firecracker does not do any specific setup for this to work. You can look at the [`fallocate`](https://www.man7.org/linux/man-pages/man2/fallocate.2.html) and see if it helps.

**Reply by pb8o** [reply]
`truncate -s SIZE` should also create a sparse file. I.e.

```sh
truncate -s 1G mydisk
mkfs.ext4 mydisk
# add some data 
truncate -s 2G mydisk
resize2fs mydisk
```
